superset/db_engine_specs/base.py

```diff
@@ -1724,6 +1726,18 @@ def get_public_information(cls) -> Dict[str, Any]:
             "disable_ssh_tunneling": cls.disable_ssh_tunneling,
         }
 
+    @classmethod
+    def validate_database_uri(cls, sqlalchemy_uri: URL) -> None:
+        """
+        Validates a database SQLAlchemy URI per engine spec.
+        Use this to implement a final validation for unwanted connection configuration
+
+        :param sqlalchemy_uri:
+        """
+        for query_param in sqlalchemy_uri.query.keys():
```

Comment:
Should we also check `connect_args`?

Reply:
Good point, but I don't think it's possible for a user to use `connect_args` on MySQL driver directly.